File: Buoi_3/Count_the_Pairs.java
package STP_2022.Buoi_3;

import java.io.FileInputStream;
import java.io.FileNotFoundException;
import java.util.Arrays;
import java.util.Scanner;

public class Count_the_Pairs {
    public static void main(String[] args) throws FileNotFoundException {
        System.setIn(new FileInputStream("/Users/khnhkd/StudySpace/java-study/src/STP_2022/input.txt"));
        Scanner scanner = new Scanner(System.in);

        int n = scanner.nextInt(), k = scanner.nextInt(), count = 0;
        int[] arr = new int[n];

        for (int i=0; i<n; i++)
            arr[i] = scanner.nextInt();

        Arrays.sort(arr);

        for (int i=0; i<n; i++)
//            if (bSearch(arr, arr[i] + k))
            if (Arrays.binarySearch(arr, arr[i] + k) >= 0)
                count++;
        System.out.println(count);
    }
}

File: Buoi_3/count_the_pairs.c
#include <stdio.h>
#include <stdlib.h>

int cmpfunc (const void * a, const void * b) {
   return ( *(int*)a - *(int*)b );
}

int main() {
    freopen("/Users/khnhkd/StudySpace/java-study/src/STP_2022/input.txt", "r", stdin);
    int n, k, compare;
    int *item;
    scanf("%d %d", &n, &k);
    int array[n];
    for (int i = 0; i < n; i++)
        scanf("%d", &array[i]);
    qsort(array, n, sizeof(int), cmpfunc);
    int count = 0;
    for (int i = 0; i < n; i++){
            compare = array[i] + k;
            item = (int*) bsearch (&compare, array, n, sizeof (int), cmpfunc);
            if( item != NULL )
                count++;
        }
    printf("%d", count);
    return 0;
}